internal/runners/activate/default.go

```diff
@@ -0,0 +1,269 @@
+			pathsEqual, err := fileutils.PathsEqual(filepath.Dir(exe.fpath), filepath.Dir(prevExe.fpath))
+			if err != nil {
+				return result, errs.Wrap(err, "Could not compare paths")
+			}
+			if !pathsEqual {
+				continue // Earlier PATH entries win
+			}
+			if funk.IndexOf(pathExt, prevExe.ext) < funk.IndexOf(pathExt, exe.ext) {
+				continue // Earlier PATHEXT entries win
+			}
+		}
+
+		exeFiles[exe.name] = exe
+	}
+
+	for _, exe := range exeFiles {
+		result = append(result, exe.fpath)
+	}
+	return result, nil
+}
+
+// shimTargetPath returns the full path to the shim target (adds .bat on Windows)
+func shimTargetPath(targetDir string, path string) string {
+	target := filepath.Clean(filepath.Join(targetDir, filepath.Base(path)))
+	if rt.GOOS != "windows" {
+		return target
+	}
+
+	oldExt := filepath.Ext(target)
+	return target[0:len(target)-len(oldExt)] + ".bat"
+}
+
+func shim(fpath, shimPath string) error {
+	exe, err := os.Executable()
+	if err != nil {
+		return errs.Wrap(err, "Could not get State Tool executable")
+	}
+
+	tplParams := map[string]interface{}{
+		"exe":     exe,
+		"command": filepath.Base(fpath),
+		"target":  fpath,
+	}
+	box := packr.NewBox("../../../assets/shim")
+	var shimStr string
+	if rt.GOOS != "windows" {
+		shimBytes := box.Bytes("shim.sh")
+		shimStr, err = strutils.ParseTemplate(string(shimBytes), tplParams)
+		if err != nil {
+			return errs.Wrap(err, "Could not parse shim.sh template")
+		}
+	} else {
+		shimBytes := box.Bytes("shim.bat")
+		shimStr, err = strutils.ParseTemplate(string(shimBytes), tplParams)
+		if err != nil {
+			return errs.Wrap(err, "Could not parse shim.bat template")
+		}
+	}
+
+	err = ioutil.WriteFile(shimPath, []byte(shimStr), 0755)
+	if err != nil {
+		return errs.Wrap(err, "failed to write shim command %s", shimPath)
+	}
+	return nil
+}
+
+func needsRollback() bool {
+	return true
+}
+
+// shimsWithTarget creates shims in the target path of all executables found in the bins dir
+// It overwrites existing files, if the overwrite flag is set.
+// On Windows the same executable name can have several file extensions,
+// therefore executables are only shimmed if it has not been shimmed for a
+// target (with the same or a different extension) from a different directory.
+// Also: Only the executable with the highest priority according to pathExt is shimmed.
+func shimsWithTarget(targetPath string, exePaths []string, out output.Outputer) error {
+	out.Print(locale.Tl("default_shim", "Writing default installation to {{.V0}}.", targetPath))
+	for _, exePath := range exePaths {
+		shimPath := shimTargetPath(targetPath, exePath)
+
+		// The link should not exist as we are always rolling back old shims before we run this code.
+		if fileutils.TargetExists(shimPath) {
+			return locale.NewInputError(
+				"err_default_symlink_target_exists",
+				"Cannot create shim as the target already exists: {{.V0}}.", shimPath)
+		}
+
+		logging.Debug("Shimming %s at %s", exePath, shimPath)
```

Comment:
This debug statement seems like it should live inside the function you're calling below